storm-core/src/jvm/org/apache/storm/executor/Executor.java

```diff
@@ -0,0 +1,567 @@
+            Map<String, IMetric> nameToRegistry = null;
+            if (taskToMetricToRegistry != null) {
+                nameToRegistry = taskToMetricToRegistry.get(taskId);
+            }
+            if (nameToRegistry != null) {
+                IMetricsConsumer.TaskInfo taskInfo = new IMetricsConsumer.TaskInfo(hostname, workerTopologyContext.getThisWorkerPort(),
+                        componentId, taskId, Time.currentTimeSecs(), interval);
+                List<IMetricsConsumer.DataPoint> dataPoints = new ArrayList<>();
+                for (Map.Entry<String, IMetric> entry : nameToRegistry.entrySet()) {
+                    IMetric metric = entry.getValue();
+                    Object value = metric.getValueAndReset();
+                    if (value != null) {
+                        IMetricsConsumer.DataPoint dataPoint = new IMetricsConsumer.DataPoint(entry.getKey(), value);
+                        dataPoints.add(dataPoint);
+                    }
+                }
+                if (!dataPoints.isEmpty()) {
+                    sendUnanchored(taskData, Constants.METRICS_STREAM_ID, new Values(taskInfo, dataPoints), executorTransfer);
+                }
+            }
+        } catch (Exception e) {
+            throw Utils.wrapInRuntime(e);
+        }
+    }
+
+    protected void setupMetrics() {
+        for (final Integer interval : intervalToTaskToMetricToRegistry.keySet()) {
+            StormTimer timerTask = (StormTimer) workerData.get("user-timer");
+            timerTask.scheduleRecurring(interval, interval, new Runnable() {
+                @Override
+                public void run() {
+                    TupleImpl tuple =
+                            new TupleImpl(workerTopologyContext, new Values(interval), (int) Constants.SYSTEM_TASK_ID, Constants.METRICS_TICK_STREAM_ID);
+                    List<AddressedTuple> metricsTickTuple = Lists.newArrayList(new AddressedTuple(AddressedTuple.BROADCAST_DEST, tuple));
+                    receiveQueue.publish(metricsTickTuple);
+                }
+            });
+        }
+    }
+
+    public void sendUnanchored(Task task, String stream, List<Object> values, ExecutorTransfer transfer) {
+        Tuple tuple = task.getTuple(stream, values);
+        List<Integer> tasks = task.getOutgoingTasks(stream, values);
+        if (tasks.size() == 0) {
+            return;
+        }
+        for (Integer t : tasks) {
+            transfer.transfer(t, tuple);
+        }
+    }
+
+    /**
+     * Send sampled data to the eventlogger if the global or component level debug flag is set (via nimbus api).
+     */
+    public void sendToEventLogger(Executor executor, Task taskData, List values, String componentId, Object messageId, Random random) {
+        Map<String, DebugOptions> componentDebug = executor.getStormComponentDebug().get();
+        DebugOptions debugOptions = componentDebug.get(componentId);
```

Comment:
can we use ternary operator as below.
DebugOptions debugOptions = (componentDebug.get(componentId) ! = null) ? componentDebug.get(componentId) : componentDebug.get(executor.getStormId());